lib/service/local_notification_service.dart

```diff
@@ -0,0 +1,218 @@
+import 'dart:convert';
+
+import 'package:flutter_local_notifications/flutter_local_notifications.dart';
+import 'package:myecl/service/class/message.dart' as message_class;
+import 'package:myecl/service/provider_list.dart';
+import 'package:myecl/tools/functions.dart';
+import 'package:qlevar_router/qlevar_router.dart';
+import 'package:rxdart/subjects.dart';
+import 'package:timezone/timezone.dart' as tz;
+import 'package:timezone/data/latest.dart' as tz;
+
+class LocalNotificationService {
+  LocalNotificationService() {
+    onNotificationClick.listen(onNotificationClickListener);
+  }
+
+  final _localNotificationService = FlutterLocalNotificationsPlugin();
+  final BehaviorSubject<message_class.Message> onNotificationClick =
+      BehaviorSubject();
+
+  Future<void> init() async {
+    tz.initializeTimeZones();
+    const AndroidInitializationSettings initializationSettingsAndroid =
+        AndroidInitializationSettings('@mipmap/launcher_icon');
+    final DarwinInitializationSettings initializationSettingsDarwin =
+        DarwinInitializationSettings(
+            onDidReceiveLocalNotification: onDidReceiveLocalNotification);
+    const LinuxInitializationSettings initializationSettingsLinux =
+        LinuxInitializationSettings(defaultActionName: 'Open notification');
+    final InitializationSettings initializationSettings =
+        InitializationSettings(
+            android: initializationSettingsAndroid,
+            iOS: initializationSettingsDarwin,
+            linux: initializationSettingsLinux);
+    _localNotificationService.initialize(initializationSettings,
+        onDidReceiveNotificationResponse: onDidReceiveNotificationResponse,
+        onDidReceiveBackgroundNotificationResponse:
+            onDidReceiveBackgroundNotificationResponse);
+  }
+
+  Future<NotificationDetails> _notificationDetails() async {
```

Comment:
Why is there a '_' only on this function ?

Reply:
To declare it as private, I might rename it